task_manager/scripts/subtask_managers/manipulation_tasks.py

```diff
@@ -62,9 +67,47 @@ def __init__(self, task_manager, task, mock_data=False) -> None:
             self.node, MoveJoints, "/manipulation/move_joints_action_server"
         )
 
+        self.gripper_client = self.node.create_client(SetDigitalIO, "/xarm/set_tgpio_digital")
+
         self._get_joints_client = self.node.create_client(GetJoints, "/manipulation/get_joints")
 
-    @service_check("move_joint_positions", -1, TIMEOUT)
+    def open_gripper(self):
+        """Opens the gripper"""
+        return self._set_gripper_state("open")
+
+    def close_gripper(self):
+        """Closes the gripper"""
+        return self._set_gripper_state("close")
+
+    def _set_gripper_state(self, state: str):
+        """
+        Controls the gripper state.
+        State: 'open' o 'close'
+        """
```

Comment:
please either use an enum or two constants to define open or close, you never know